libs/components/molecules/src/lib/interactive-svg/interactive-svg.component.ts

```diff
@@ -106,6 +116,48 @@ export class InteractiveSvgComponent<T extends NodeMapEntry> implements OnDestro
   /** Destroys */
   private destroy$ = new Subject<void>();
 
+  /** Crosswalk element of svg */
+  private crosswalkEl?: Element;
+
+  /**
+   * Updates the highlighting based on current highlight id
+   * @param changes
+   */
+  ngOnChanges(changes: SimpleChanges): void {
+    if ('highlightId' in changes && this.crosswalkEl) {
+      this.resetHighlight();
+      const cellMatch = this.mapping?.find((entry) => entry.id === this.highlightId); // find the matching cell data in mapping
+      let crosswalkMatch: NodeListOf<Element>;
+      if (cellMatch) {
+        const name = cellMatch.name;
+        const childId = name.replace(/_/g, '_x5F_'); // get the id of the matching cell in svg
+        const elementMatch = this.crosswalkEl.querySelectorAll(`#${childId}`)[0]; //find the element of the matching cell in svg
+        if (elementMatch) {
+          if (elementMatch.nodeName === 'g') {
+            // highlights children if g element
+            crosswalkMatch = this.crosswalkEl.querySelectorAll(`#${childId} :is(path, polygon, polyline)`);
+          } else {
+            // highlights siblings if path, polygon, or polyline element
+            const parentId = elementMatch.parentElement?.id ?? '';
+            crosswalkMatch = this.crosswalkEl.querySelectorAll(`#${parentId} :is(path, polygon, polyline)`);
+          }
+          crosswalkMatch.forEach((element) =>
```

Comment:
It would be better to add a class to each element rather than setting styles. That way the styles can be easily changed in the scss file. Then you would remove the class on resetHightlight